### PR TITLE
Backporting various off heap issues to 9.1.x

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-8409
https://issues.jboss.org/browse/ISPN-8419
https://issues.jboss.org/browse/ISPN-8481
https://issues.jboss.org/browse/ISPN-8423
https://issues.jboss.org/browse/ISPN-8454

### CODE REVIEW

File: core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public BoundedOffHeapDataContainer(int desiredSize, long maxSize, EvictionType t
       if (type == EvictionType.COUNT) {
          sizeCalculator = i -> 1;
       } else {
-         // Use size of entry plus 28 for our LRU pointer node
+         // Use size of entry plus 16 for our LRU pointers
          sizeCalculator = i -> offHeapEntryFactory.getSize(i) + OffHeapLruNode.getSize();
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java]: Doesn't `offHeapEntryFactory.getSize()` include the size of the LRU pointers?

Comment (wburns) [reply core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java]: Ah yes it very well might now that I consolidated them into a single block.

**Comment by wburns** [comment]
Updated so memory count is correct again.

**Comment by wburns** [comment]
Please also make sure that changes from https://github.com/infinispan/infinispan/pull/5577 are cherry picked into this one after this integrated.

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Pull request merged**